airbyte-webapp/src/components/ui/PillSelect/PillSelect.tsx

```diff
@@ -34,7 +36,7 @@ export const PillSelect: React.FC<PillSelectProps> = ({ className, ...props }) =
                 active={isOpen}
                 className={className}
               >
-                {label}
+                {disabled ? <FormattedMessage id="connectionForm.bulkEdit.pillButtonLabel.notAvailable" /> : label}
```

Comment:
This formatted message is part of the connection form, but this component is part of the UI, so it crosses different domains. We're not sure if the default behavior for disabled is always to say "not available"

Reply:
@edmundito I added `renderDisabledState: () => React.ReactNode` property for `PillSelect`, so now, when `disabled: true`, the content can be controlled from different application parts. please let me know if this looks good to you